src/nvim/option.c

```diff
@@ -3221,6 +3184,31 @@ did_set_string_option (
   return errmsg;
 }
 
+// 'matchpairs' is a comma-separated string of
+// [(multibyte) char]:[(multibyte) char]
+// for example "(:),[:],{:}"
+static bool mps_valid(char_u *mps)
+{
+  for (char_u *p = mps; *p != NUL; p++) {
+    p += mb_ptr2len(p);
+
+    if (*p != ':') {
+      return false;
+    }
+    p++;
+
+    if (*p == NUL) {
+      return false;
+    }
+    p += mb_ptr2len(p);
+
+    if (*p != NUL && *p != ',') {
+      return false;
+    }
+  }
```

Comment:
Did I comment here to remove the `NUL` check? The check cannot be removed if `p++` is used in the loop header. For `"(:)"` the loop would check for `NUL` one byte after the C string.